pkg/kubernetes/kubernetes.go

```diff
@@ -205,6 +207,23 @@ func (h *Client) LabelNodeAsMaster(name string) (err error) {
 
 	n.Labels[constants.LabelNodeRoleMaster] = ""
 
+	taintFound := false
+
+	for _, taint := range n.Spec.Taints {
+		if taint.Key == constants.LabelNodeRoleMaster && taint.Value == "true" {
+			taintFound = true
+			break
+		}
+	}
+
+	if !taintFound {
+		n.Spec.Taints = append(n.Spec.Taints, corev1.Taint{
+			Key:    constants.LabelNodeRoleMaster,
+			Value:  "true",
+			Effect: corev1.TaintEffectNoSchedule,
+		})
+	}
+
```

Comment:
As an alternative, we could pass in the `--register-with-taints` flag to the kubelet: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/. We would just need to be mindful that user's could set that flag in the `extraArgs` field, and we would have to merge them. Not sure I care either way, but wanted to point this option out to see what you think.

Reply:
you confused me now, now I'm stuck in [Buridan's ass](https://en.wikipedia.org/wiki/Buridan%27s_ass) paradox

Reply:
:laughing: First time seeing that one. Love it. Let's stick with this. It is easier to merge this toleration in with any user supplied ones this way.